src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/source_dependant_iframes/MixinLivingEntity.java

```diff
@@ -0,0 +1,62 @@
+package com.unascribed.fabrication.mixin.g_weird_tweaks.source_dependant_iframes;
+
+import com.unascribed.fabrication.interfaces.TickSourceIFrames;
+import com.unascribed.fabrication.support.EligibleIf;
+import com.unascribed.fabrication.support.MixinConfigPlugin;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Mixin(LivingEntity.class)
+@EligibleIf(configAvailable="*.source_dependant_iframes")
+public abstract class MixinLivingEntity extends Entity implements TickSourceIFrames {
+
+	public MixinLivingEntity(EntityType<?> type, World world) {
+		super(type, world);
+	}
+
+	Set<String> fabrication$iframeTracker = new HashSet<>();
+	int fabrication$timeUntilRegen = 0;
+
+	@Inject(at=@At("HEAD"), method="damage(Lnet/minecraft/entity/damage/DamageSource;F)Z")
+	private void checkDependentIFrames(DamageSource source, float amount, CallbackInfoReturnable<Boolean> cir) {
+		if (!MixinConfigPlugin.isEnabled("*.source_dependant_iframes")) return;
+		if (fabrication$iframeTracker.add(source.getName())) {
```

Comment:
Maybe specialcase indirect damage sources by adding the name of the originator to the key? This way multiple entities can deal damage at the same time